File: library/src/main/java/me/proxer/library/api/media/MediaApi.java
package me.proxer.library.api.media;

import retrofit2.Retrofit;

/**
 * Api for the Media class.
 *
 * @author Ruben Gees
 */
public final class MediaApi {

    private final InternalApi internalApi;

    /**
     * Only for internal use.
     */
    public MediaApi(final Retrofit retrofit) {
        this.internalApi = retrofit.create(InternalApi.class);
    }

    /**
     * Returns the respective endpoint.
     */
    public CalendarEndpoint calendar() {
        return new CalendarEndpoint(internalApi);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/media/MediaApi.kt
package me.proxer.library.api.media

import retrofit2.Retrofit

/**
 * Api for the Media class.
 *
 * @author Ruben Gees
 */
class MediaApi internal constructor(retrofit: Retrofit) {

    private val internalApi = retrofit.create(InternalApi::class.java)

    /**
     * Returns the respective endpoint.
     */
    fun calendar(): CalendarEndpoint {
        return CalendarEndpoint(internalApi)
    }
}
